pkg/api/validate.go

```diff
@@ -44,6 +44,34 @@ func isAzureZone(fqdn string) bool {
 	return strings.HasSuffix(fqdn, ".cloudapp.azure.com") && len(strings.Split(fqdn, ".")) == 5
 }
 
+func isValidIPV4CIDR(cidr string) bool {
+	ip, net, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return false
+	}
+	if ip.To4() == nil {
+		return false
+	}
+	if net == nil || !ip.Equal(net.IP) {
+		return false
+	}
+	return true
+}
+
+func vnetContainsSubnet(vnet, subnet *net.IPNet) bool {
+	if !vnet.IP.Equal(subnet.IP.Mask(vnet.Mask)) {
+		return false
+	}
+
+	vnetsize, _ := vnet.Mask.Size()
+	subnetsize, _ := subnet.Mask.Size()
+	if vnetsize > subnetsize {
```

Comment:
Should this be the opposite, i.e. `vnetsize < subnetsize` then `false` should be returned?

Reply:
I think this is correct, but it's not particularly clear.  Maybe I'll s/size/bits/ and add a comment.
Say the vnet is 10.0.0.0/16 and the subnet is 10.0.0.0/8.  The vnet does not contain the subnet, so we should return false.  Then vnetsize is 16, and subnetsize is 8.  16 > 8, therefore we return false.

Reply:
Also there are unit test cases for this.

Reply:
Yes, agreed, I read this as `vnetSize` but it's actually VNET mask size so this check makes sense.